readthedocs/builds/models.py

```diff
@@ -132,6 +133,37 @@ def __str__(self):
             ),
         )
 
+    @property
+    def ref(self):
+        if self.slug == STABLE:
+            stable = determine_stable_version(self.project.versions.all())
+            if stable:
+                return stable.slug
+
+    @property
+    def vcs_url(self):
+        url = ''
+        if self.slug == STABLE:
+            slug_url = self.ref
+        elif self.slug == LATEST:
+            slug_url = self.project.default_branch or self.project.vcs_repo().fallback_branch
+        else:
+            slug_url = self.slug
+
+        if self.project.repo_type in ('git', 'gitlab'):
```

Comment:
`repo_type` means what I expects. Here we need to know what kind of repository it is to generate a valid http linkfor the online VCS for that version. 

This method should probably need to be refactored, anyway. For now, it does what I need, but it's not robust and does not work for all the cases.

Reply:
Oh, you both are right!

I got confused here. I want to check if `github` or `gitlab` or `bitbucket` is in the repository's URL since I'm trying to build the URL for the version on the external VCS service.